### DISCUSSION TITLE
grafana stopping

### DISCUSSION THREAD

**Discussion by configmanCooper** [cmu-sei/GHOSTS#263] [General]
I know there is documentation about chowning g_data but I have done that

I'm just trying to set up following the directions here:
mkdir ghosts-api
cd ghosts-api
curl https://raw.githubusercontent.com/cmu-sei/GHOSTS/master/src/Ghosts.Api/docker-compose.yml -o docker-compose.yml
docker-compose up -d

I'm in the latest version of Ubuntu
The error I'm getting is as follows:

(HTTP code 500) server error - error while creating mount source path '/host_mnt/home/configuration/grafa/datasources': mkdir /host_mnt/home/configuration: permission denied

This error is in docker desktop.

I've tried chowning the g_data directory to the current user, to root, etc. I've created that directory myself with mkdir. I've changed the user in the yml. But I can't start the grafana container. It immediately stops and gives the above error

**Comment by configmanCooper** [comment]
Found the fix with other people that run into this issue 

Change the yaml file to the following

grafana:
    image: grafana/grafana
    container_name: ghosts-grafana
    depends_on:
      - postgres
    ports:
      - "3000:3000"
    networks:
      - ghosts-network
    restart: always
    volumes:
      - grafana-storage:/var/lib/grafana

volumes:
  grafana-storage:


Credit to https://cylab.be/blog/71/simulate-user-activity-with-the-ghosts-framework-introduction#:
for the issue resolution

**Comment by sei-dupdyke** [comment]
Hi sorry for the delay in looking at this — crazy week — and thanks for checking out the project.

This part of the new yaml proposed can't work:

```yaml
volumes:
- grafana-storage:/var/lib/grafana

volumes:
grafana-storage:
```

unless there is more that was clipped off (or I am missing something).

But I do think there is something about running the container as root that is problematic on systems. So I am removing that line (and TBH, I can't recall why we put it there to begin with), [per this commit](https://github.com/cmu-sei/GHOSTS/commit/a3e75311bb6b537a86f2be4dfcd5ecaf84a03936).

Thanks for troubleshooting this one, appreciate it!

**Reply by pezhore** [reply]
I think the formatting was cut off from the other user's proposal.

This should work:

```
volumes:
  - grafana-storage:/var/lib/grafana

volumes:
  grafana-storage:
```
